examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -172,11 +153,39 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
 
             this._migrationP = undefined;
 
+            this.taskManager.off("lost", exitMigration);
+            this.taskManager.off("completed", exitMigration);
+            this.taskManager.off("assigned", onAssigned);
+            this.taskManager.complete(this.migrationTaskId);
+
+            this.takeAppropriateActionForCurrentMigratable();
+        };
+
+        const exitMigration = (taskId: string) => {
+            if (taskId !== this.migrationTaskId) {
+                return;
+            }
+            this.taskManager.off("lost", exitMigration);
+            this.taskManager.off("completed", exitMigration);
+            this.taskManager.off("assigned", onAssigned);
+            migratedModel?.close();
```

Comment:
I don't think it's necessary to close the migratedModel here (at least in all cases) - e.g. if I lose connection for a second and so lose the task, the migratedModel I'm holding is still good.